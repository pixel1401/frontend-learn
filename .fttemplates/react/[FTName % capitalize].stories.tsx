import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ThemeDecorator } from 'shared/config/storybook/ThemeDecorator/ThemeDecorator';
import { Theme } from 'app/providers/ThemeProvider';
import { <FTName | capitalize> } from './<FTName>';

export default {
    title: 'any/<FTName | capitalize>',
    component: <FTName | capitalize>,
    argTypes: {
        backgroundColor: { control: 'color' },
    },
} as ComponentMeta<typeof <FTName | capitalize>>;

const Template: ComponentStory<typeof <FTName | capitalize>> = (args) => <<FTName | capitalize> {...args} />;

export const Primary = Template.bind({});
Primary.args = {
};

export const Dark = Template.bind({});
Dark.args = {
};
Dark.decorators = [ThemeDecorator(Theme.DARK)];
