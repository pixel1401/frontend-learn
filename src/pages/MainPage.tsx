import NavBar from "src/components/NavBar";

 const MainPage = () => {
    return (
        <>
            <h1>MainPAge</h1>
            {/* <NavBar/> */}
        </>
    )
};

export default MainPage;