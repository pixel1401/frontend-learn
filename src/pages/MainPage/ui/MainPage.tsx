import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { getFetch } from 'shared/lib/tests/fetch';

const MainPage = () => {
    const { t, i18n } = useTranslation();

    const start = async () => {
        console.log(await getFetch());
    };

    useEffect(() => {
        start();
    }, []);

    return (
        <div>
            {t('MainPage')}
            <button type="button">Hello..</button>
        </div>
    );
};

export default MainPage;
