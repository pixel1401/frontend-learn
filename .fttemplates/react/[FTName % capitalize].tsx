import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import cls from './<FTName>.module.scss';

interface <FTName | capitalize>Props {}

export const <FTName | capitalize> : FC<<FTName | capitalize>Props> = () => {
    const { t } = useTranslation();

    return (
        <div>
        </div>
    );
};
