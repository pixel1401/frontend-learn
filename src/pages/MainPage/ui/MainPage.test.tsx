import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import MainPage from './MainPage';
// import { MainPage } from 'pages/MainPage';

test('main page Test btn', () => {
    render(<MainPage />);
    const button = screen.getByText(/MainPage/i);
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot('new');
});
