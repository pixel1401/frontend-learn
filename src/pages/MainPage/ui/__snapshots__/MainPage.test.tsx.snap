// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main page Test btn 1`] = `
<div>
  MainPage
  <button
    type="button"
  >
    Hello
  </button>
</div>
`;

exports[`main page Test btn: new 1`] = `
<div>
  MainPage
  <button
    type="button"
  >
    Hello..
  </button>
</div>
`;
